src/inference/gibbs_conditional.jl

```diff
@@ -0,0 +1,135 @@
+"""
+    GibbsConditional(sym, conditional)
+
+A "pseudo-sampler" to manually provide analytical Gibbs conditionals to `Gibbs`.
+`GibbsConditional(:x, cond)` will sample the variable `x` according to the conditional `cond`, which
+must therefore be a function from a `NamedTuple` of the conditioned variables to a `Distribution`.
+
+
+The `NamedTuple` that is passed in contains all random variables from the model in an unspecified
+order, taken from the [`VarInfo`](@ref) object over which the model is run.  Scalars and vectors are
+stored as their respective types, but all as floats.  The tuple also contains the value of the
```

Comment:
Does the implementation require the limitation to floats?
```suggestion
stored as their respective types, but all as floats. The tuple also contains the value of the
```

Reply:
I believed it's a limitation of `VarInfo`, at least the untyped one -- is that wrong?

Reply:
The untyped `VarInfo` saves everything in a `Vector{Real}`: https://github.com/TuringLang/DynamicPPL.jl/blob/9431df05af76460b0d5fea9572fcaaf946b4791d/src/varinfo.jl#L160

Reply:
OK, it probably is type-preserving with TypeVarInfo, but maybe not with the untyped one.  I removed the wording, it should rarely be a problem I guess.